src/sqlparser/src/parser.rs

```diff
@@ -121,6 +122,12 @@ pub struct Parser {
     /// Since we cannot distinguish `>>` and double `>`, so use `angle_brackets_num` to store the
     /// number of `<` to match `>` in sql like `struct<v1 struct<v2 int>>`.
     angle_brackets_num: i32,
+    /// It's important that already in named Array or not. so use this field check in or not.
+    /// Consider 0 is you're not in named Array. if more than 0 is you're in named Array
+    array_depth: i32,
+    /// We cannot know current array should be keep named or not, so by using this field store
+    /// every depth of array that should be keep named or not.
+    array_named_map: HashMap<i32, bool>,
```

Comment:
Are these essentially a stack of bools? Note that entries of this HashMap are inserted by first sibling and removed by parent after last sibling, and we only look at the value of the current layer.

Reply:
```
[array[1,2], array[3,4]]
```
Because I thought when enter to `parse_array_expr` I can only know current `named` state. not past state. So I made a `array_named_map` for past state. 
and I want to put together all logic about array to this function. so I think this is the best I can do. (I thought...?) if there's another way then I needed.

Reply:
I meant, `HashMap` is an overkill here as we do not need random access. A stack (`Vec<bool>`) is more natural here.

* `hash_map::Entry::Vacant(entry) = self.array_named_map.entry(self.peek_array_depth())` -> `self.peek_array_depth() > stack.len()`
* `self.array_named_map.contains_key(&ended_array_depth)` -> `self.peek_array_depth() < stack.len()`
* `self.array_named_map.get(&self.peek_array_depth())` -> `stack.last()`
* btw, `array_depth` can be `usize`

Correct me if I missed anything.